api/admin/controller.py

```diff
@@ -228,12 +228,23 @@ def edit(self, data_source, identifier):
             )
             changed = True
 
-        new_summary = flask.request.form.get("summary")
+        new_summary = flask.request.form.get("summary") or ""
         if new_summary != work.summary_text:
+            old_summary = None
+            if work.summary and work.summary.data_source == staff_data_source:
+                old_summary = work.summary
+
             (link, is_new) =  work.primary_edition.primary_identifier.add_link(
                 Hyperlink.DESCRIPTION, None, 
                 staff_data_source, content=new_summary)
             work.set_summary(link.resource)
+
```

Comment:
That's what the code below this is for. If the work's previous summary has the staff data source, it and its link get deleted.